addons/mail/static/src/models/message/message.js

```diff
@@ -442,6 +442,32 @@ function factory(dependencies) {
                 this.originThread.isModeratedByCurrentPartner
             );
         }
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsSubjectEqualThreadName() {
+            if (!this.subject || !this.originThreadName) {
+                return false;
+            }
+
+            const threadName = this.originThreadName.toLowerCase().trim();
+            let prefix = ['re:', 'fw:', 'fwd:'];
+            let cleanedSubject = this.subject.toLowerCase();
+            let startingSubject;
+            while (cleanedSubject !== startingSubject) {
+                startingSubject = cleanedSubject;
+                for (const pref of prefix) {
+                    if (cleanedSubject.startsWith(pref)) {
+                        cleanedSubject = cleanedSubject.replace(pref, '').trim();
+                        break;
+                    }
+                }
+                if (threadName === startingSubject) {
```

Comment:
Shouldn't this compare `cleanedSubject` to `threadName` instead? I'm trying to find a case where this would lead to an incorrect value to add yet another test but I don't have it right now.

Actually having `startingSubject` is confusing, I guess instead we could just have a boolean to control the loop, for example `wasSubjectCleaned`, that is initially `true`, that we reset to `false` at the start of every loop, and that we set to `true` again before the `break`.